File: sdkwork-backend-sdk-python/sdkwork_backend_sdk/api/chat_message.py
from typing import Any, Dict, List, Optional
from ..http_client import HttpClient
from ..models import ChatMessageQueryListForm, PlusApiResultBoolean, PlusApiResultListPlusChatMessageContentVO, PlusApiResultListPlusChatMessageVO, PlusApiResultPagePlusChatMessageContentVO, PlusApiResultPagePlusChatMessageVO, PlusApiResultPlusChatMessageContentVO, PlusApiResultPlusChatMessageVO, PlusChatMessageContentForm, PlusChatMessageForm, QueryListForm

class ChatMessageApi:
    """chat_message API client."""
    
    def __init__(self, client: HttpClient):
        self._client = client

    def update(self, body: PlusChatMessageContentForm) -> PlusApiResultPlusChatMessageContentVO:
        """Update an existing chat message"""
        return self._client.put(f"/backend/v3/api/message", json=body)

    def create(self, body: PlusChatMessageContentForm) -> PlusApiResultPlusChatMessageContentVO:
        """Create a new chat message"""
        return self._client.post(f"/backend/v3/api/message", json=body)

    def update_message(self, body: PlusChatMessageForm) -> PlusApiResultPlusChatMessageVO:
        """Update an existing chat message"""
        return self._client.put(f"/backend/v3/api/chat/message", json=body)

    def create_message(self, body: PlusChatMessageForm) -> PlusApiResultPlusChatMessageVO:
        """Create a new chat message"""
        return self._client.post(f"/backend/v3/api/chat/message", json=body)

    def list_by_page(self, body: Optional[QueryListForm] = None, params: Optional[Dict[str, Any]] = None) -> PlusApiResultPagePlusChatMessageContentVO:
        """Get chat messages by page"""
        return self._client.post(f"/backend/v3/api/message/list", json=body, params=params)

    def list_all_entities(self, body: Optional[QueryListForm] = None) -> PlusApiResultListPlusChatMessageContentVO:
        """Get all chat messages"""
        return self._client.post(f"/backend/v3/api/message/list/all", json=body)

    def load_more(self, body: Optional[ChatMessageQueryListForm] = None, params: Optional[Dict[str, Any]] = None) -> PlusApiResultPagePlusChatMessageVO:
        """Get chat messages by page"""
        return self._client.post(f"/backend/v3/api/chat/message/load_more", json=body, params=params)

    def create_list_by_page(self, body: Optional[ChatMessageQueryListForm] = None, params: Optional[Dict[str, Any]] = None) -> PlusApiResultPagePlusChatMessageVO:
        """Get chat messages by page"""
        return self._client.post(f"/backend/v3/api/chat/message/list", json=body, params=params)

    def create_list_all_entities(self, body: Optional[ChatMessageQueryListForm] = None) -> PlusApiResultListPlusChatMessageVO:
        """Get all chat messages"""
        return self._client.post(f"/backend/v3/api/chat/message/list/all", json=body)

    def get_by_id(self, id: str) -> PlusApiResultPlusChatMessageContentVO:
        """Get chat message by ID"""
        return self._client.get(f"/backend/v3/api/message/{id}")

    def delete(self, id: str) -> PlusApiResultBoolean:
        """Delete a chat message"""
        return self._client.delete(f"/backend/v3/api/message/{id}")

    def get_by_id_message(self, id: str) -> PlusApiResultPlusChatMessageVO:
        """Get a chat message by ID"""
        return self._client.get(f"/backend/v3/api/chat/message/{id}")

    def delete_message(self, id: str) -> PlusApiResultBoolean:
        """Delete a chat message"""
        return self._client.delete(f"/backend/v3/api/chat/message/{id}")

File: sdkwork-backend-sdk-flutter/lib/src/api/chat_message.dart
import '../http/client.dart';
import '../models.dart';

class ChatMessageApi {
  final HttpClient _client;
  
  ChatMessageApi(this._client);

  /// Update an existing chat message
  Future<PlusApiResultPlusChatMessageContentVO?> update(PlusChatMessageContentForm body) async {
    final response = await _client.put(ApiPaths.backendPath('/message'), body: body, contentType: 'application/json');
    return response is PlusApiResultPlusChatMessageContentVO ? response : null;
  }

  /// Create a new chat message
  Future<PlusApiResultPlusChatMessageContentVO?> create(PlusChatMessageContentForm body) async {
    final response = await _client.post(ApiPaths.backendPath('/message'), body: body, contentType: 'application/json');
    return response is PlusApiResultPlusChatMessageContentVO ? response : null;
  }

  /// Update an existing chat message
  Future<PlusApiResultPlusChatMessageVO?> updateMessage(PlusChatMessageForm body) async {
    final response = await _client.put(ApiPaths.backendPath('/chat/message'), body: body, contentType: 'application/json');
    return response is PlusApiResultPlusChatMessageVO ? response : null;
  }

  /// Create a new chat message
  Future<PlusApiResultPlusChatMessageVO?> createMessage(PlusChatMessageForm body) async {
    final response = await _client.post(ApiPaths.backendPath('/chat/message'), body: body, contentType: 'application/json');
    return response is PlusApiResultPlusChatMessageVO ? response : null;
  }

  /// Get chat messages by page
  Future<PlusApiResultPagePlusChatMessageContentVO?> listByPage(QueryListForm? body, Map<String, dynamic>? params) async {
    final response = await _client.post(ApiPaths.backendPath('/message/list'), body: body, params: params, contentType: 'application/json');
    return response is PlusApiResultPagePlusChatMessageContentVO ? response : null;
  }

  /// Get all chat messages
  Future<PlusApiResultListPlusChatMessageContentVO?> listAllEntities(QueryListForm? body) async {
    final response = await _client.post(ApiPaths.backendPath('/message/list/all'), body: body, contentType: 'application/json');
    return response is PlusApiResultListPlusChatMessageContentVO ? response : null;
  }

  /// Get chat messages by page
  Future<PlusApiResultPagePlusChatMessageVO?> loadMore(ChatMessageQueryListForm? body, Map<String, dynamic>? params) async {
    final response = await _client.post(ApiPaths.backendPath('/chat/message/load_more'), body: body, params: params, contentType: 'application/json');
    return response is PlusApiResultPagePlusChatMessageVO ? response : null;
  }

  /// Get chat messages by page
  Future<PlusApiResultPagePlusChatMessageVO?> createListByPage(ChatMessageQueryListForm? body, Map<String, dynamic>? params) async {
    final response = await _client.post(ApiPaths.backendPath('/chat/message/list'), body: body, params: params, contentType: 'application/json');
    return response is PlusApiResultPagePlusChatMessageVO ? response : null;
  }

  /// Get all chat messages
  Future<PlusApiResultListPlusChatMessageVO?> createListAllEntities(ChatMessageQueryListForm? body) async {
    final response = await _client.post(ApiPaths.backendPath('/chat/message/list/all'), body: body, contentType: 'application/json');
    return response is PlusApiResultListPlusChatMessageVO ? response : null;
  }

  /// Get chat message by ID
  Future<PlusApiResultPlusChatMessageContentVO?> getById(String id) async {
    final response = await _client.get(ApiPaths.backendPath('/message/${id}'));
    return response is PlusApiResultPlusChatMessageContentVO ? response : null;
  }

  /// Delete a chat message
  Future<PlusApiResultBoolean?> delete(String id) async {
    final response = await _client.delete(ApiPaths.backendPath('/message/${id}'));
    return response is PlusApiResultBoolean ? response : null;
  }

  /// Get a chat message by ID
  Future<PlusApiResultPlusChatMessageVO?> getByIdMessage(String id) async {
    final response = await _client.get(ApiPaths.backendPath('/chat/message/${id}'));
    return response is PlusApiResultPlusChatMessageVO ? response : null;
  }

  /// Delete a chat message
  Future<PlusApiResultBoolean?> deleteMessage(String id) async {
    final response = await _client.delete(ApiPaths.backendPath('/chat/message/${id}'));
    return response is PlusApiResultBoolean ? response : null;
  }
}
